code/Infrastructure/Verticals/Arr/SonarrClient.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Text;
+using Common.Configuration;
+using Domain.Sonarr;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Infrastructure.Verticals.Arr;
+
+public sealed class SonarrClient : ArrClient
+{
+    public SonarrClient(ILogger<SonarrClient> logger, IHttpClientFactory httpClientFactory)
+        : base(logger, httpClientFactory)
+    {
+    }
+
+    public override async Task RefreshItemsAsync(ArrInstance arrInstance, HashSet<int> itemIds)
```

Comment:
Maybe not for now, but when all arr support is implemented i am pretty sure some of this method code can be extracted into the parent and then just call the base in the children. altho i think the base should be called after the custom code which is not the way it is recommended.

Reply:
I agree that more improvements can be done with the code and I'll look into that.
I don't think calling `base.BaseMethod()` is a requirement. Also for this particular method, the base method is abstract.

Reply:
well, i was thinking about moving part of the code into the parent (such as the try-catch block) and that would make the base.baseMethod call required.